core/dbt/parser/manifest.py

```diff
@@ -1280,6 +1287,17 @@ def _check_resource_uniqueness(
 
         alias_resources[full_node_name] = node
 
+    intersection_versioned = set(versioned_resources.keys()).intersection(
+        set(unversioned_resources.keys())
+    )
+    if intersection_versioned:
```

Comment:
I think raising this sets us back in terms of being able to handle ref-able resource with duplicate names across packages. For example - if I define project_a.model (unversioned) and project_b.model.v1 (versioned) - those work when run in their respective projects as root, but when installed as packages would raise `DuplicateVersionedUnversionedError`

Reply:
I'm thinking that we need to store the versioned_resources and unversioned_resources by package. Maybe we'd only need to check the root package?

Reply:
I've updated the code to check versioned/unversioned by packages and tweaked the error message to also suggest adding the unversioned model to the versions.